monai/deploy/operators/monai_bundle_inference_operator.py

```diff
@@ -151,6 +156,45 @@ def _extract_from_archive(
     if isinstance(config_names, str):
         config_names = [config_names]
 
+    # Check if bundle_path is a directory (for directory-based bundles)
+    bundle_path_obj = Path(bundle_path)
+    if bundle_path_obj.is_dir():
```

Comment:
Separating out the "reading configs from file folder" to a helper function would be much better, just one line to return the result from calling it.